app/assets/javascripts/discourse/app/components/edit-category-topic-template.js

```diff
@@ -1,12 +1,59 @@
 import { buildCategoryPanel } from "discourse/components/edit-category-panel";
 import { observes } from "discourse-common/utils/decorators";
 import { schedule } from "@ember/runloop";
+import FormTemplate from "admin/models/form-template";
+import I18n from "I18n";
+import { action, computed } from "@ember/object";
+import Yaml from "js-yaml";
 
 export default buildCategoryPanel("topic-template", {
   // Modals are defined using the singleton pattern.
   // Opening the insert link modal will destroy the edit category modal.
   showInsertLinkButton: false,
 
+  init() {
```

Comment:
I understand doing this in `init()` is not the best approach, but I've tried creating a select-kit addon and having the request happen upon click, but I ran into some issues with the label not rendering due to a promise being returned. In any case, we need to show the template content as well, so we need all the data here anyways, so I'm leaving it with this for now to get the core functionality and can revisit this to improve this later.